src/query/models/config.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package models
+
+import (
+	"errors"
+	"fmt"
+)
+
+var validIDSchemes = []IDSchemeType{
+	TypeLegacy,
+	TypeQuoted,
+	TypePrependMeta,
+}
+
+func (t IDSchemeType) validateIDSchemeType() error {
```

Comment:
Maybe worth just make this an exported method?  i.e. `Validate() error` so that if we use it outside of YAML parsing somewhere we can call validate on it?

Reply:
Sure, sounds good.

More of a general style question, but is it better to leave functions unexported initially, then export when needed, or to have "expected" behaviour like `Validate()` public even if it's not used anywhere? I guess leaving it private can lead to accidentally rewriting functionality considering how go has functions floating around everywhere in a package :p